docs/plugins/regrass.rst

```diff
@@ -5,15 +5,101 @@ regrass
+
+Options
+-------
+
+``-m``, ``--max``
+    Maxes out every grass type in the tile, giving extra grazing time.
+    Not normal DF behavior. Tile will appear to be the first type of grass
+    present in the map block until that is depleted, moving on to the next
+    type. When this option isn't used, non-depleted grass tiles will have their
+    existing type refilled, while grass-depleted soils will have a type
+    selected randomly.
+``-n``, ``--new``
+    Adds biome-compatible grass types that were not originally present in the
+    map block. Allows regrass to work in blocks that never had any grass to
+    begin with. Will still fail in incompatible biomes.
+``-f <grass_id>``, ``--force <grass_id>``
+    Force a grass type on tiles with no compatible grass types. ``grass_id`` is
+    not case-sensitive, but must be enclosed in quotes if spaces exist. If an
+    empty string is used via ``""``, then a single random grass type will be
+    selected from the world's raws. The ``--new`` option takes precidence for
+    compatible biomes, otherwise such tiles will be forced instead.
+``-a``, ``--ashes``
+    Regrass tiles that've been burnt to ash.
+``-d``, ``--buildings``
+    Regrass tiles under certain passable building tiles including stockpiles,
+    planned buildings, workshops, and farms. (Farms will convert grass tiles to
+    furrowed soil after a short while, but is useful with ``--mud`` option.)
+    Doesn't work on "dynamic" buildings such as doors and floor grates.
+    (Dynamic buildings also include levers and cage traps, for some reason.)
+    Existing grass tiles will always be refilled, regardless of building tile.
+``-u``, ``--mud``
+    Converts non-smoothed, mud-spattered stone into grass. Valid for layer
+    stone, obsidian, and ore.
+``-b``, ``--block``
+    Only regrass the map block that contains the first ``pos`` argument.
+    ``devel/block-borders`` can be used to visualize map blocks.
+``-z``, ``--zlevel``
+    Regrass entire z-levels. Will do all z-levels between ``pos`` arguments if
+    both given, z-level of first ``pos`` if one given, else z-level of
+    viewscreen if no ``pos`` given.
+
+Examples
+--------
+
+``regrass``
+    Regrass the entire map, refilling existing and depleted grass.
+``regrass here``
+    Regrass the selected tile, refilling existing and depleted grass.
+``regrass here 0,0,90 --zlevel``
+    Regrass all z-levels including the selected tile's z-level through z-level
+    90, refilling existing and depleted grass.
+``regrass 0,0,100 19,19,119 --ashes --mud``
+    Regrass tiles in the 20x20x20 cube defined by the coords, refilling
+    existing and depleted grass, and converting ashes and muddy stone (if
+    respective blocks ever had grass.)
+``regrass 10,10,100 -baudnm``
+    Regrass the block that contains the given coord; converting ashes, muddy
+    stone, and tiles under buildings; adding all compatible grass types, and
+    filling each grass type to max.
+``regrass -f ""``
```

Comment:
I realize this was borne from the lack of optional optargs, but I'm not a fan of this syntax. This is too novel compared to other tools.

How about if `-f`, `--force` takes zero parameters and a separate `--grass-type=<type>` allows the player to (optionally) specify the grass type?